paddle/framework/operator.h

```diff
@@ -281,51 +281,47 @@ class OpKernel {
 
 class OperatorWithKernel : public OperatorBase {
  public:
-  struct OpKernelKey {
-    platform::Place place_;
-
-    OpKernelKey() = default;
-    explicit OpKernelKey(const platform::DeviceContext& dev_ctx) {
-      place_ = dev_ctx.GetPlace();
-    }
-
+  struct OpKernelKey : public platform::Place {
```

Comment:
I think we need to consider it, this OpKernelKey is used to find right kernel according to RunContext, now we register kernel like 
```cpp
REGISTER_OP_CPU_KERNEL(add_two,
                       ops::AddKernel<paddle::platform::CPUPlace, float>);
```
because all our kernel is with type `float`, it has no problem now, but if we register
```cpp
REGISTER_OP_CPU_KERNEL(add_two,
                       ops::AddKernel<paddle::platform::CPUPlace, float16>);
```
this two kernels will overwrite each other. I can send a PR to fix this today.